WBooks/Book Details/BookDetailsViewController.swift

```diff
@@ -36,47 +36,56 @@ class BookDetailsViewController: UIViewController {
     
     private func setBookDetails() {
         _view.setBook(bookViewModel: _viewModel.bookViewModel)
-        _viewModel.loadRents(onSuccess: onLoadRentsSuccess)
-    }
-    
-    func onLoadRentsSuccess(rents: [Rent]) {
-        _viewModel.onLoadRentsSuccess(rents: rents)
-        _view.setAvailability(status: _viewModel.status)
+        _viewModel.status.producer.startWithValues { [weak self] status in
+            guard let this = self else {
+                return
+            }
+            this._view.setAvailability(status: status)
+        }
+        _viewModel.loadRents()
     }
     
     func setRentButton() {
+        //Rent button action
         _view.rent.reactive.controlEvents(.touchUpInside).observeValues { [weak self] _ in
             self?.rent()
         }
+        
+        //Actions when rent request is completed
+        _viewModel.rentState.producer.startWithValues { [weak self] state in
+            guard let this = self else {
+                return
+            }
+            switch state {
+            case .rentError:
```

Comment:
this enum seems to be "equal" to add enum.. you could have only one generic enum if they have the same cases